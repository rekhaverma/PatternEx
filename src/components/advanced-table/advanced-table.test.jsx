import React from 'react';
import AdvancedTable from './advanced-table.component';

/**
 * @status: Complete
 * @sign-off-by: Bogdan Iacoboae
 */
describe('<AdvancedTable />', () => {
  beforeEach(() => {
    global.localStorage = jest.fn();
    global.localStorage.setItem = jest.fn();
    global.localStorage.getItem = jest.fn();
  });
  it('should match with snapshot', () => {
    const props = {
      children: <div />,
      rowHeight: 300,
      maxPageSize: '5',
      rowHighlightIndex: 1,
      pageSize: '5',
      classname: 'test',
      tableConfig: [
        {
          className: 'table__headerColumn',
          field: 'hash',
          hidden: true,
          intl: 'malicious.table.hash',
          isKey: true,
          label: 'Hash',
          resizable: false,
        },
        {
          className: 'table__headerColumn',
          field: 'start_time_formatted',
          hidden: false,
          intl: 'malicious.table.start_time_formatted',
          isKey: false,
          label: 'Log Date',
          resizable: true,
          width: 100,
        }],
      data: [{
        hash: 'asd-test',
        start_time_formatted: 'start-time',
      }],
      onRowClick: jest.fn(),
      locationPage: '@tEST/LOCATION',
    };


    const tree = shallow(<AdvancedTable {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
