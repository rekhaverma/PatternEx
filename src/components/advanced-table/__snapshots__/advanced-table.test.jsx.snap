// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdvancedTable /> should match with snapshot 1`] = `
<div
  className="table-section"
>
  <div
    className="table-top-section"
  >
    <div />
    <WithBoxHOC
      activeOptions={
        Array [
          "start_time_formatted",
        ]
      }
      options={
        Array [
          Object {
            "content": "Hash",
            "disabled": false,
            "id": "hash",
          },
          Object {
            "content": "Log Date",
            "disabled": false,
            "id": "start_time_formatted",
          },
        ]
      }
      updateOptionsHandler={[Function]}
    />
  </div>
  <ConfigurableTable
    classname="test"
    data={
      Array [
        Object {
          "hash": "asd-test",
          "start_time_formatted": "start-time",
        },
      ]
    }
    locationPage="@tEST/LOCATION"
    maxPageSize="5"
    onRowClick={[Function]}
    pageSize="5"
    rowHeight={300}
    rowHighlightIndex={1}
    tableConfig={
      Array [
        Object {
          "className": "table__headerColumn",
          "field": "hash",
          "hidden": true,
          "intl": "malicious.table.hash",
          "isKey": true,
          "label": "Hash",
          "resizable": false,
        },
        Object {
          "className": "table__headerColumn",
          "field": "start_time_formatted",
          "hidden": false,
          "intl": "malicious.table.start_time_formatted",
          "isKey": false,
          "label": "Log Date",
          "resizable": true,
          "width": 100,
        },
      ]
    }
  >
    <div />
  </ConfigurableTable>
</div>
`;
